metadata/src/main/java/org/apache/kafka/controller/ControllerMetricsManager.java

```diff
@@ -202,10 +202,10 @@ private void handleFencingChange(Integer brokerId, BrokerRegistrationFencingChan
             throw new IllegalArgumentException(String.format("Broker with id %s is not registered", brokerId));
         }
 
-        if (fencingChange == BrokerRegistrationFencingChange.FENCE) {
+        if (fencingChange.equals(BrokerRegistrationFencingChange.FENCE)) {
             fencedBrokers.add(brokerId);
             updateBrokerStateMetrics();
-        } else if (fencingChange == BrokerRegistrationFencingChange.UNFENCE) {
+        } else if (fencingChange.equals(BrokerRegistrationFencingChange.UNFENCE)) {
```

Comment:
This and the line above are not needed since they're enums, right?

Reply:
Yes. These cahnges are not needed.